src/components/_layout/Header/Header.tsx

```diff
@@ -7,26 +7,33 @@ import { SrOnlySpan } from '@/styles'
 import { useAuthorization } from '@/providers'
 
 import { UserMenu } from '@/components/UserMenu'
+import ArrowCircleLeftIcon from '@mui/icons-material/ArrowCircleLeft'
+import { useLocation } from 'react-router-dom'
 
 export const Header = () => {
   const { authorization } = useAuthorization()
-  const authorizationStateAsBoolean = !!authorization
+  const isAuthorized = !!authorization
+
+  const location = useLocation()
+  const backButtonEnabled = location.pathname !== '/' && location.pathname !== '/dashboard'
 
   return (
     <HeaderStyled>
       <Row>
         <Column>
-          <HeaderContent authorizationState={authorizationStateAsBoolean}>
-            <LogoLink
-              to={authorization ? '/dashboard' : '/'}
-              authorizationState={authorizationStateAsBoolean}
-              variant="transparent"
-              isLink
-            >
+          <HeaderContent isAuthorized={isAuthorized}>
+            <LogoLink to={isAuthorized ? '/dashboard' : '/'} isAuthorized={isAuthorized} variant="transparent" isLink>
+              {backButtonEnabled && (
+                <>
+                  <SrOnlySpan>Go back to the main page</SrOnlySpan>
+                  <LogoBackButton>
+                    <ArrowCircleLeftIcon fontSize="large" />
+                  </LogoBackButton>
+                </>
+              )}
               <Logo src="/logo-text.png" alt="spv wallet logo" width="250" height="53" />
-              <SrOnlySpan>Go back to the main page</SrOnlySpan>
             </LogoLink>
```

Comment:
I've removed the `BackToRootButton`, but instead I've implemented a global mechanism for displaying "back icon" with the logo. It shows it where a user is not at the "root" page. 
![backButton](https://github.com/bitcoin-sv/spv-wallet-web-frontend/assets/152964795/5975d7a2-0e1f-4892-b489-f01cfc73a03f)